lib/rules/background-color-validation/index.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @author Aleksandra Borowska <https://github.com/Ola2808-Boro>
+ */
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const isStandardSyntaxDeclaration = require('../../utils/isStandardSyntaxDeclaration');
+const ruleName = 'background-color-validation';
+
+const messages = ruleMessages(ruleName, {
+	rejected:
+		`Don't use this color as background-color property. Level of energy-consuming in OLED's screen is depend at luminance of color.\n` +
+		`If you want to create more sustainable website use colors with lower luminance`,
+});
+
+const meta = {
+	url: 'http://ec0lint.com/features/background-color-validation',
+};
+
+/** @type {import('ec0lint-style').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls((decl) => {
+			if (
+				!decl.prop.toString().includes('background-color') &&
+				!isStandardSyntaxDeclaration(decl)
+			) {
+				return;
+			}
+
+			const message = ruleMessages(ruleName, {
+				rejected:
+					`Don't use this color as background-color property. Level of energy-consuming in OLED's screen is depend at luminance of color.\n` +
+					`If you want to create more sustainable website use colors with lower luminance`,
+			});
+
+			const Colors = {
+				white: 'white',
+				blue: 'blue',
+				aque: 'aque',
+				magenta: 'magenta',
+				lime: 'lime',
+				cream: 'cream',
+				yellow: 'yellow',
+				violet: 'violet',
+				red: 'red',
+				orange: 'orange',
+			};
+
+			//convert HEX to RGB
+			if (decl.value.match(/^#([\da-f]{6}){1,2}$/i)) {
```

Comment:
let's export this to a separate funtion (inside the same file) so that the code will be cleaner

Reply:
or use constants/variables defined in one place, e.g. const colorRegex = { hex: ..., rgb: ... } - then it will be clear what's checked and where